lib/active_model/serializer.rb

```diff
@@ -112,17 +109,30 @@ def self.attributes(*attrs)
     #     enr
     def self.attribute(attr, options = {})
       key = options.fetch(:key, attr)
-      _attributes_keys[attr] = { key: key } if key != attr
-      _attributes << key unless _attributes.include?(key)
+      _attributes_map[key] = attr
 
       ActiveModelSerializers.silence_warnings do
-        define_method key do
+        define_method attr do
           object.read_attribute_for_serialization(attr)
-        end unless method_defined?(key) || _fragmented.respond_to?(attr)
+        end unless method_defined?(attr) || _fragmented.respond_to?(attr)
       end
     end
 
     # @api private
+    # An accessor for the old _attributes internal API
+    def self._attributes
+      _attributes_map.keys
+    end
+
+    # @api private
+    # An accessor for the old _attributes_keys internal API
+    def self._attributes_keys
```

Comment:
can you provide an example in the method comment about what this method does (before and after).

This is to help future developers debug more painlessly